src/pages/Home/index.tsx

```diff
@@ -1,14 +1,40 @@
 import { Container } from 'react-bootstrap';
-
+import { useEffect, useState } from 'react';
+import { OrderProps, BobaProps } from 'types/common/main';
+import { bobaList } from 'assets/sampleBobaAPI';
 import BobaContainer from 'components/BobaContainer';
-import { OrderProps } from 'types/common/main';
+import SearchBar from 'components/SearchBar';
 
 const Homepage = (props: OrderProps): JSX.Element => {
   const { order, setOrder } = props;
+
+  const [bobaListx, setBobaListx] = useState<BobaProps[]>(bobaList);
+  const [searchQuery, setSearchQuery] = useState<string>('');
+
+  const filteredfunction = (bobaList: BobaProps[], query: string): BobaProps[] => {
```

Comment:
Rename the filteredfunction function to something more descriptive, such as filterBobaList:

Reply:
Use the useMemo hook to memoize the filtered boba list, so that it only gets recalculated when either the boba list or the search query changes:
```
const filteredBobaList = useMemo(() => filterBobaList(bobaList, searchQuery), [bobaList, searchQuery]);
```

Reply:
i am confused about this use , can you elaborate ?

Reply:
It's just alternative. We can keep as it is now.